include/nbl/builtin/hlsl/emulated/float64_t.hlsl

```diff
@@ -0,0 +1,657 @@
+        /*static emulated_float64_t create(uint16_t val)
+        {
+            return emulated_float64_t(bit_cast<uint64_t>(float64_t(val)));
+        }*/
+
+        NBL_CONSTEXPR_STATIC_INLINE this_t create(this_t val)
+        {
+            return val;
+        }
+
+        NBL_CONSTEXPR_STATIC_INLINE this_t create(int32_t val)
+        {
+            return bit_cast<this_t>(emulated_float64_t_impl::castToUint64WithFloat64BitPattern(int64_t(val)));
+        }
+
+        NBL_CONSTEXPR_STATIC_INLINE this_t create(int64_t val)
+        {
+            return bit_cast<this_t>(emulated_float64_t_impl::castToUint64WithFloat64BitPattern(val));
+        }
+
+        NBL_CONSTEXPR_STATIC_INLINE this_t create(uint32_t val)
+        {
+            return bit_cast<this_t>(emulated_float64_t_impl::castToUint64WithFloat64BitPattern(uint64_t(val)));
+        }
+
+        NBL_CONSTEXPR_STATIC_INLINE this_t create(uint64_t val)
+        {
+            return bit_cast<this_t>(emulated_float64_t_impl::castToUint64WithFloat64BitPattern(val));
+        }
+
+        NBL_CONSTEXPR_STATIC_INLINE this_t create(float32_t val)
+        {
+            this_t output;
+            output.data = emulated_float64_t_impl::castFloat32ToStorageType<FlushDenormToZero>(val);
+            return output;
+        }
+
+        NBL_CONSTEXPR_STATIC_INLINE this_t create(float64_t val)
+        {
+#ifdef __HLSL_VERSION
+            emulated_float64_t retval;
+            uint32_t lo, hi;
+            asuint(val, lo, hi);
+            retval.data = (uint64_t(hi) << 32) | uint64_t(lo);
+            return retval;
+#else
+            return bit_cast<this_t>(val);
+#endif
+        }
+
+        // arithmetic operators
+        this_t operator+(const emulated_float64_t rhs) NBL_CONST_MEMBER_FUNC
+        {
+            // TODO: remove
+            float64_t sum = bit_cast<float64_t>(data) + bit_cast<float64_t>(rhs.data);
+            uint64_t sumAsUint = bit_cast<uint64_t>(sum);
+
+            this_t output2;
+            output2.data = sumAsUint;
+
+            return output2;
+
+            if (FlushDenormToZero)
+            {
+                if(!FastMath)
+                {
+                    if (tgmath::isNaN(data) || tgmath::isNaN(rhs.data))
+                        return bit_cast<this_t>(ieee754::traits<float64_t>::quietNaN);
+
+                    if (emulated_float64_t_impl::areBothInfinity(data, rhs.data))
+                    {
+                        uint64_t lhsSign = data & ieee754::traits<float64_t>::signMask;
+                        uint64_t rhsSign = rhs.data & ieee754::traits<float64_t>::signMask;
+
+                        if (lhsSign == rhsSign)
+                            return bit_cast<this_t>(ieee754::traits<float64_t>::inf | lhsSign);
+                        else if (lhsSign || rhsSign)
+                            return bit_cast<this_t>(ieee754::traits<float64_t>::quietNaN | ieee754::traits<float64_t>::signMask);
+                    }
```

Comment:
you should have a general function that checks for both infinities having the same signs